cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -41,40 +43,52 @@ def domain(request):
     return dom
 
 
-@pytest.fixture(scope="module")
-def test_vm(setup_provider_modscope, provider, vm_name, request):
-    """Fixture to provision appliance to the provider being tested if necessary"""
-    vm = VM.factory(vm_name, provider, template_name=provider.data['full_template']['name'])
+def provision_vm(provider, vm_name, request, small_template=True):
+    if small_template:
+        vm = VM.factory(vm_name, provider, template_name=provider.data['small_template'])
```

Comment:
```small_template``` doesn't follow the same format and have a ```name``` attribute? That seems poor. (not blaming you for this). Would be nice to fix this up so it looks like
```python
template_source = 'small_template' if small_template else 'full_template':
vm = VM.factory(vm_name, provider, template_name=provider.data[template_source]['name'])
```